server/src/main/java/org/elasticsearch/repositories/RepositoriesModule.java

```diff
@@ -48,7 +49,11 @@ public RepositoriesModule(
         RecoverySettings recoverySettings,
         TelemetryProvider telemetryProvider
     ) {
+        // TODO: refactor APM metrics into their own class, passed in as a dependancy.
```

Comment:
I do agree with you to refactor this out into their own class, but I'd rather do it in a separate PR.

Reply:
Yeah, it may involve changing a bunch of the tests, cluttering these changes up. Agreed.

Reply:
FYI, I have an ongoing PR https://github.com/elastic/elasticsearch/pull/102505 that adds more this registration. I can take up the task to refactor them after both PRs are done.

Reply:
Thanks @ywangd 🙂